litmus-portal/graphql-server/pkg/database/mongodb/workflow/operations.go

```diff
@@ -95,6 +95,22 @@ func GetWorkflows(query bson.D) ([]ChaosWorkFlowInput, error) {
 	return workflows, nil
 }
 
+// GetWorkflowByID takes a workflow ID to retrieve the workflow details from the database
+func GetWorkflowByID(workflowID string) (*ChaosWorkFlowInput, error) {
+	ctx, _ := context.WithTimeout(backgroundContext, 10*time.Second)
+	result, err := mongodb.Operator.Get(ctx, mongodb.WorkflowCollection, bson.D{{"workflow_id", workflowID}})
+	if err != nil {
+		return nil, err
+	}
+	var workflow ChaosWorkFlowInput
+	err = result.Decode(&workflow)
+	if err != nil {
+		return nil, err
+	}
+
+	return &workflow, nil
+}
+
```

Comment:
Can we re-use the existing `GetWorkflow(query bson.D)` function for this instead of creating a new one? You can directly pass the query required in this one and get the required workflow.

Reply:
Sure thing, didn't pay attention to the function just after this one :man_facepalming: